ui/app/templates/permissions.html

```diff
@@ -46,29 +55,91 @@
 
-<div class="panel panel-default" style="clear:right"
-     ng-repeat="user in filtered_items = (users | filter:filterBySearch) | orderBy:ordering | startFrom:(currentPage - 1)*pageSize | limitTo:pageSize">
-  <div class="panel-heading">
+      <h3 class="panel-title">
+        <div style="float: right">
+          <i ng-show="username && $first">Current</i>
+          <button ng-show="!username" class="btn btn-default btn-xs" ng-click="openNewScheduledPermissionChangeModal(user)">Schedule a Change</button>
+          <button ng-show="!username" class="btn btn-default btn-xs" ng-click="openUpdateModal(user)">Update</button>
+        </div>
+        <span title="Username" ng-bind-html="highlightSearch(user.username, 'username')"></span>
+      </h3>
+    </div>
 
-    <h3 class="panel-title">
-      <div style="float: right">
-        <i ng-show="username && $first">Current</i>
-        <button ng-show="!username" class="btn btn-default btn-xs" ng-click="openNewScheduledPermissionChangeModal(user)">Schedule a Change</button>
-        <button ng-show="!username" class="btn btn-default btn-xs" ng-click="openUpdateModal(user)">Update</button>
-      </div>
-      <span title="Username" ng-bind-html="highlightSearch(user.username, 'username')"></span>
-    </h3>
   </div>
 
+
+  <div class="pagination-container" ng-show="users_filtered_items.length > pageSize">
+    <pagination class="pagination-sm" total-items="users_filtered_items.length" ng-model="currentPage" items-per-page="pageSize"></pagination>
+  </div>
 </div>
 
+<!-- Roles tab begins here -->
+<div ng-show="isSet(2)">
+<div class="row">
+  <div class="col-sm-6">
+    <h2>
+      <span ng-show="!username">Roles</span>
+      <span ng-show="username">{{ username }}</span>
 
-<div class="pagination-container" ng-show="filtered_items.length > pageSize">
-  <pagination
-    class="pagination-sm"
-    total-items="filtered_items.length"
-    ng-model="currentPage"
-    items-per-page="pageSize"
-  ></pagination>
+      <span class="count">
+    ({{ roles_list.length | number:0 }}
+    <span ng-bind="hasFilter() ? 'found' : 'in total'"></span>)
+      </span>
+
+      <button class="btn btn-primary btn-xs" ng-show="!username" ng-click="openNewModal()">
+    Add a new permission
+    <i class="glyphicon glyphicon-plus"></i>
+  </button>
+    </h2>
+  </div>
+
+  <div class="col-sm-5 text-right role-dropdown">
+    <div class="row" ng-show="!rule_id">
+      <label>All Roles:</label>
+      <select id="role-select-box" ng-model="current_role" name="current_role" ng-options="current_role for current_role in roles_list">
+      </select>
+    </div>
+  </div>
 </div>
+
+  <div class="pagination-container" ng-show="roles_list.length > pageSize">
+    <pagination class="pagination-sm" total-items="roles_list.length" ng-model="currentPage" items-per-page="pageSize"></pagination>
+  </div>
+
+  <div ng-if="isAllRoles(current_role)" class="panel-group" ng-repeat="(key, role) in roles_users">
```

Comment:
This is another place where inlining would help readability - there's no point in calling out to the controller for this.